noir/noir-repo/aztec_macros/src/transforms/compute_note_hash_and_nullifier.rs

```diff
@@ -49,20 +49,20 @@ pub fn inject_compute_note_hash_and_nullifier(
     crate_id: &CrateId,
     context: &mut HirContext,
 ) -> Result<(), (AztecMacroError, FileId)> {
-    if let Some((module_id, file_id)) = get_contract_module_data(context, crate_id) {
+    if let Some((_, module_id, file_id)) = get_contract_module_data(context, crate_id) {
         // If compute_note_hash_and_nullifier is already defined by the user, we skip auto-generation in order to provide an
         // escape hatch for this mechanism.
         // TODO(#4647): improve this diagnosis and error messaging.
-        if check_for_compute_note_hash_and_nullifier_definition(crate_id, context) {
+        if context.crate_graph.root_crate_id() != crate_id
```

Comment:
ah just like in the storage case, we don't want the imported contracts' notes to end up polluting our `compute_note_hash_and_nullifier`.
Does this work with notes imported from libraries (e.g. from aztec-nr)? I'd guess so because the notes are imported/used in the root crate?

Reply:
It does but is in fact very tricky, because we still want to pull things like `ValueNote` from `EasyPrivateState` for example. It's is a little bit fragile in some edge cases and I want to improve it.